### DISCUSSION TITLE
FastAPI Background Tasks/Asyncio Working Locally but not on Vercel

### DISCUSSION THREAD

**Discussion by araavp** [vercel/community#1276] [Help]
### Summary

I have a React/FastAPI app. Locally, with either Background Tasks or Asyncio, the methods are able to run another function in the background while sending a response to the request right away (regardless of whether the background task finished). See example below:

```
@app.get("/{user}")
def func(user, background_tasks: BackgroundTasks) -> dict:
        try:
            background_tasks.add_task(func2, user)
        except:
            return {"data": ""}
    return {"data": "func2 is running"}
```
   

However, when I deploy to Vercel, with either of the methods, func2 runs synchronously and the func method does not return right away. This is a problem as func2 takes much longer than the request timeout. Does anyone know any fixes?

**Comment by HEVpro** [comment]
Hi! I'm on the same problem. When I look the logs in vercel it shows how starts my background task step by step. :(

**Comment by PrivFelixKniest** [comment]
Janurary 2024 and I still have the same issue, did you guys find any fix to this?

**Comment by devraj** [comment]
Being a long time Python/FastAPI developer, I assume this is largely because Vercel doesn't run the equivalent of a `uvicorn` which `starlette` and `fastapi` would rely onto execute background tasks. 

I have been looking into background tasks / queues in general on vercel and the general consensus seems to be using a third party provider like inngest.

**Reply by PrivFelixKniest** [reply]
I went over to render.com aswell and the bootup time is annoying but its alright for my usecase, thanks a lot for answering, I will have a look at it!
